accounts-journey/presentation/viewmodel/AccountsListViewModel.swift

```diff
@@ -8,52 +8,124 @@
-    }
+    @Published var allAccounts = [AccountUIModel]()
+    @Published var screenState: ScreenState = .loading
     
     // MARK: - Private
     
-    private lazy var accountsUsecase: AccountsUseCase = {
-        guard let usecase = Resolver.optional(AccountsUseCase.self) else {
+    private lazy var accountsUseCase: AccountsUseCase = {
+        guard let useCase = Resolver.optional(AccountsUseCase.self) else {
             fatalError("AccountsUseCase needed to continue")
         }
-        return usecase
+        return useCase
     }()
     
-    private var cancellables = Set<AnyCancellable>()
-    private let output: PassthroughSubject<Output, Never> = .init()
     
-    func transform(input: AnyPublisher<Input, Never>) -> AnyPublisher<Output,Never> {
-        input.sink {[weak self] event in
-            switch event {
-            case .viewDidAppear, .viewDidRefresh:
-                self?.getAccountSummary()
-            }
-        }.store(in: &cancellables)
-        
-        return output.eraseToAnyPublisher()
+    func onEvent(_ event: AccountListEvent) {
+        switch event {
+        case .getAccounts:
+            getAccountSummary(fromEvent: .getAccounts)
+        case .refresh:
+            getAccountSummary(fromEvent: .refresh)
+        case .search(let searchString):
+            getAccountSummary(fromEvent: .search(searchString))
+        }
     }
     
-    private func getAccountSummary() {
+    func getAccountSummary(fromEvent event: AccountListEvent) {
+        var query = ""
+        
+        if case let .search(searchString) = event {
+            query = searchString
+        }
+        
+        screenState  = .loading
         
-        accountsUsecase.getAccountSummary {[weak self] result in
+        accountsUseCase.getAccountSummary {[weak self] result in
+            guard let self else {
+                return
+            }
+            
             switch result {
             case let .success(accountsSummaryResponse):
-                self?.output.send(.fetchDidSucceed(accountSummary: accountsSummaryResponse))
+                allAccounts = accountsSummaryResponse
+                    .toMapUI()
+                    .generateList(query: query)
+                
+                if allAccounts.isEmpty {
+                    screenState = .emptyResults(
+                        stateViewConfiguration(
+                            for: .noAccounts)
+                    )
+                } else {
+                    screenState = .loaded
+                }
+                
             case let .failure(errorResponse):
-                self?.output.send(.fetchDidFail(error: errorResponse.error!))
+                screenState = .hasError(
+                    stateViewConfiguration(
+                        for: .loadingFailure(errorResponse)
+                    )
+                )
             }
         }
     }
     
 }
+
+extension AccountsListViewModel: UITableViewDataSource {
```

Comment:
ViewModel doesn't know anything about the UI. The two extensions are very UI related. Is it possible to move the extensions to the UI class?
I think the code is equivalent of the ListAdapter in Android: https://github.com/Backbase/golden-sample-app-android/blob/main/accounts-journey/src/main/kotlin/com/backbase/accounts_journey/presentation/ui/AccountListAdapter.kt

Reply:
I agree but, there's a thin line here, just because the protocol is name `UITableViewDataSource`. 

But if you look at the methods being called, within extension, these are you specialised to providing the data to the table view and there's nothing "view" related.

Also, from a second look this can be moved to the UI layer thereby depend on the `allAccounts` property from the `viewModel` thus make it look like it's on it's own.